packages/lesswrong/components/common/MixedTypeFeed.tsx

```diff
@@ -165,9 +169,19 @@ export const MixedTypeFeed = <
   useEffect(maybeStartLoadingMore);
   useOnPageScroll(maybeStartLoadingMore);
   
-  const results = (data && resolverName && data[resolverName]?.results) || [];
+  const results = useMemo(() => 
+    (data && resolverName && data[resolverName]?.results) || [],
+    [data, resolverName]
+  );
   const orderPolicy = reorderOnRefetch ? 'no-reorder' : undefined;
   const orderedResults = useOrderPreservingArray(results, keyFunc, orderPolicy);
+  
+  // Call onDataLoaded when loading state changes
+  useEffect(() => {
+    if (onDataLoaded) {
+      onDataLoaded(results, loading);
```

Comment:
If this is only supposed to be called when the data is actually loaded (as the docstring for the field suggests), this isn't right, but since you're passing in `loading` I assume it handles all the possible cases?